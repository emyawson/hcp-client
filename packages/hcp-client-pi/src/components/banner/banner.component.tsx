import * as React from 'react';
import { withTheme } from 'styled-components';

import { BannerDiv, BannerP } from './banner.style';

interface BannerProps {
  theme: any;
  children?: any;
}

const BannerComponent = (props: BannerProps) => {
  const { children } = props;

  return (
    <BannerDiv>
      <BannerP>{children}</BannerP>
    </BannerDiv>
  );
};

export const Banner = withTheme(BannerComponent);
