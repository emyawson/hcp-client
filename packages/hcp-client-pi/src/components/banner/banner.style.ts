import styled, { IThemeInterface } from '@roche/patterns-indicators/theme';
import { StyledComponentClass } from 'styled-components';

interface BannerDivProps {
  backgroundColor?: string;
}

export const BannerDiv: StyledComponentClass<BannerDivProps, IThemeInterface> = styled<BannerDivProps, 'div'>('div')`
  background-color: ${props => props.theme.colors.blueMarine};
  border-radius: ${props => props.theme.borderRadius[1]};
  color: ${props => props.theme.colors.white};
  padding: 1em ${props => props.theme.spacing.four};
  text-align: center;
  width: 100%;
`;

export const BannerP: StyledComponentClass<{}, IThemeInterface> = styled<{}, 'p'>('p')`
  font-weight: ${props => props.theme.fontWeights.bold};
  line-height: 1;
  margin: 0 auto;
`;
