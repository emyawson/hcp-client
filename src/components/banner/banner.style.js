import styled from 'styled-components';

import { borderRadius, colors, spacing, transitions } from 'src/core';

import { P, weight } from '../fonts';

export const BannerDiv = styled.div`
  background-color: ${colors.blueMarine};
  border-radius: ${borderRadius.three};
  color: ${colors.white};
  padding: 1em ${spacing.four};
  text-align: center;
  width: 100%:
`;

export const BannerP = P.extend`
  font-weight: ${weight.bold};
  line-height: 1;
  margin: 0 auto;

  a,
  a:visited {
    color: inherit;
    transition: ${transitions.default};
    text-decoration: underline;

    &:hover {
      color: ${colors.transparentBlack};
    }
  }
`;
