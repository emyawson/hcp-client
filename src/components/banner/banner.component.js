import React from 'react';

import { BannerDiv, BannerP } from './banner.style';

export const Banner = ({ text }) => (
  <BannerDiv>
    <BannerP>{text}</BannerP>
  </BannerDiv>
);
