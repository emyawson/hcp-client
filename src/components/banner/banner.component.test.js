import React from 'react';
import { shallow } from 'enzyme';

import { Banner } from './banner.component';

describe('Banner', () => {
  const mockProps = {
    text: 'Alert! You are running a test.',
  };
  it('renders correctly', () => {
    const wrapper = shallow(<Banner {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
