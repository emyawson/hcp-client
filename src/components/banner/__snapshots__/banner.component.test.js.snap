// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    text="Alert! You are running a test."
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.p>
        Alert! You are running a test.
      </styled.p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "Alert! You are running a test.",
      },
      "ref": null,
      "rendered": "Alert! You are running a test.",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.p>
          Alert! You are running a test.
        </styled.p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Alert! You are running a test.",
        },
        "ref": null,
        "rendered": "Alert! You are running a test.",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
